pkg/routeagent_driver/handlers/kubeproxy/node_handler.go

```diff
@@ -73,9 +63,9 @@ func (kp *SyncHandler) populateRemoteVtepIps(vtepIP string, operation Operation)
 	}
 
 	logger.V(log.DEBUG).Infof("populateRemoteVtepIps is called with vtepIP %s, isGatewayNode %t",
-		vtepIP, kp.isGatewayNode)
+		vtepIP, kp.State().IsOnGateway())
 
-	if kp.isGatewayNode {
+	if kp.State().IsOnGateway() {
```

Comment:
This suggests a small race here, but it’s protected by the lock. However to make the correctness more obvious it would be better to store the current return value of `IsOnGateway()` and use that both for logging and decision-marking (so it’s apparent that the branch taken matches the log message).

Reply:
sure, for readability.